Api/Data/SearchItemResultsInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Snowdog\Menu\Api\Data;
+
+use Magento\Framework\Api\SearchResultsInterface;
+
+/**
+ * Search results interface.
+ *
+ * @api
+ */
+interface SearchItemResultsInterface extends SearchResultsInterface
```

Comment:
Shouldn't be `Snowdog\Menu\Api\Data\MenuSearchResultsInterface` ? Remember about **di.xml**